android/app/src/main/java/com/zulipmobile/notifications/NotifyReact.kt

```diff
@@ -93,11 +93,7 @@ internal fun notifyReact(application: ReactApplication, data: Bundle) {
             // initialNotification again, but it will see a notificationOpened event.
             emit(reactContext, "notificationOpened", Arguments.fromBundle(data))
     }
-    when (appStatus) {
-        null, ReactAppStatus.NOT_RUNNING, ReactAppStatus.BACKGROUND ->
-            launchMainActivity(application as Context)
-        ReactAppStatus.FOREGROUND -> Unit
```

Comment:
I didn't find any reasons given for the conditional; possibly it's
    an unquestioned relic of the Wix library we used to use.

Indeed the behavior goes back to the Wix library. Looks like it's basically unchanged since 9eddca888e33f6ca8f828582a927cbfe981f2bcc:

    android notif: Use standard upstream API for watching RN lifecycle.
    
    This has exactly the same effect as the 80 lines of code in the wix
    library's ReactAppLifecycleFacade.

with the logic a bit refactored in 9b0157d828b94226cca34004b5be2bb5a76afeb0 and 0feb940fb0af4bd0626200e5caba4d4b7a313a87.

I don't think having the conditional is necessarily bad, given that launching isn't going to do anything useful when already in the foreground. But with the changes upcoming in this branch, it becomes useful to do this unconditionally, so it's a good change now.